### DISCUSSION TITLE
[Question][Hw3] About zext instruction: How to get the operand llvalue?

### DISCUSSION THREAD

**Discussion by GetRusty** [prosyslab-classroom/cs424-program-reasoning#280] [Q&A]
Name: Ingyu Baeg, 백인규

Dear All,
I am close to implementing all the tasks in HW3 but one problem gets in my way:
When I try get the operand of the zext insturction by Llvm.operand (instr) 0, some weird llvalue ( declare dso_local i32 @input(...) #1 )
is returned and causes errors.

In the README.md file, the example llvm code is written in the convention of
%conv = zext %cmp
but in my local test directory the example llvm code is generated like this:
%conv = zext i1 %cmp to i32

So I thought, maybe I should call Llvm.operand (instr) 1 according to the index, which then causes a segmentation fault.

Is there any other way to get the operand of the zext instruction, or am I missing something on my side and doing undefined sort of behavior?

Thanks for reading this question.

![image](https://github.com/prosyslab-classroom/cs492-program-reasoning/assets/42976401/e1774c95-1cf8-4dad-8fc0-5bdc09833bde)

**Comment by goodtaeeun** [comment]
Hi.
The operand of `zext` instruction can be obtained with `Llvm.operand (instr) 0`.
I am not sure how you get `declare dso_local i32 @input(...` with `Llvm.operand (instr) 0`, when `instr` is `%conv = zext i1 %cmp to i32`.

**Comment by GetRusty** [comment]
> Hi. The operand of `zext` instruction can be obtained with `Llvm.operand (instr) 0`. I am not sure how you get `declare dso_local i32 @input(...` with `Llvm.operand (instr) 0`, when `instr` is `%conv = zext i1 %cmp to i32`.

After debugging I found that I was doing some undefined behavior in the iterations of the block, now I'm getting valid instructions for Llvm.operand (instr) 0. Thank you for the response!
